cli/node/example_test.go

```diff
@@ -0,0 +1,101 @@
+package node
+
+import (
+	"fmt"
+	"os"
+
+	"go.dedis.ch/dela/cli"
+)
+
+func ExampleCLIBuilder_Build() {
+	builder := NewBuilder(exampleController{})
+
+	cmd := builder.SetCommand("bye")
+
+	cmd.SetFlags(cli.StringFlag{
+		Name:  "name",
+		Usage: "set the name",
+		Value: "Bob",
+	})
+
+	// This action is only executed on the CLI process. It is also possible to
+	// call commands on the daemon after it has been started with "start".
+	cmd.SetAction(func(flags cli.Flags) error {
+		fmt.Printf("Bye, %s!", flags.String("name"))
+		return nil
+	})
+
+	app := builder.Build()
+
+	err := app.Run([]string{os.Args[0], "bye", "--name", "Alice"})
+	if err != nil {
+		panic("app failed: " + err.Error())
+	}
+
+	// Output: Bye, Alice!
+}
+
+// Hello is an example of a component that can be injected and resolved on the
+// daemon side.
+type Hello interface {
+	SayTo(name string)
+}
+
+type simpleHello struct{}
+
+func (simpleHello) SayTo(name string) {
+	fmt.Printf("Hello, %s!", name)
+}
```

Comment:
this logic is not demonstrated in the example though and is just dead (example) code, or am I missing something?
I'm a tiny bit confused

Reply:
To demonstrate, I need to start a daemon and that's quite complex in the example as it should be an independent command but I wanted to show how to write the controller.

Reply:
Fair enough :)